tests/steamship_tests/agents/test_context.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from steamship import Block, Steamship
+from steamship.agents.schema import Action, AgentContext
+from steamship.agents.tools.search import SearchTool
+
+
+@pytest.mark.usefixtures("client")
+def test_context_persist_hydrate(client: Steamship):
+
+    ctx = AgentContext.get_or_create(client=client, context_keys={"id": "test-foo"})
+
+    ctx.chat_history.append_user_message("this is a test yo")
+    ctx.completed_steps.append(
+        Action(
+            tool=SearchTool(),
+            input=[Block(text="where's waldo?")],
+            output=[Block(text="location unknown")],
+        )
+    )
+
+    file_handle = ctx.persist_to_file(client=client)
+    retrieved_ctx = AgentContext.hydrate_from_file(client=client, file_handle=file_handle)
```

Comment:
This feels like something that would be great to have completely independent of streaming tbh.